src/GraphWebhooks-Core/Controllers/NotificationController.cs

```diff
@@ -117,9 +118,17 @@ private async Task GetChangedMessagesAsync(IEnumerable<Notification> notificatio
                 if (notification.ResourceData.ODataType != "#Microsoft.Graph.Message") continue;
 
                 SubscriptionStore subscription = subscriptionStore.GetSubscriptionInfo(notification.SubscriptionId);
+                                
+                // Set the claims for ObjectIdentifier and TenantId, and              
+                // use the above claims for the current HttpContext
+                HttpContext.User = ClaimsPrincipalExtension.FromObjectIdAndTenantId(subscription.UserId, subscription.TenantId);
+
+                // Fetch the access token
+                string accessToken = await tokenAcquisition.GetAccessTokenOnBehalfOfUser(
+                        HttpContext, new[] { Infrastructure.Constants.ScopeMailRead });
 
-                // Initialize the GraphServiceClient. This sample uses the tenant ID the cache key.
-                GraphServiceClient graphClient = sdkHelper.GetAuthenticatedClient(subscription.TenantId);
+                // Initialize the GraphServiceClient. 
+                var graphClient = GraphServiceClientFactory.GetAuthenticatedGraphClient(accessToken);
```

Comment:
If this is a long running process (and therefore if the token risks to expire), instead of passing the token, you might want to pass a delegate to GetAuthenticatedGraphClient, which would get the token. This way the graph client would call it when needed

the call would become:

```CSharp
var graphClient = GraphServiceClientFactory.GetAuthenticatedGraphClient(() => {wait tokenAcquisition.GetAccessTokenOnBehalfOfUser(
                        HttpContext, new[] { Infrastructure.Constants.ScopeMailRead })} );
```

Reply:
```suggestion
                var graphClient = await GraphServiceClientFactory.GetAuthenticatedGraphClient(async () =>
            {
               string result = await tokenAcquisition.GetAccessTokenOnBehalfOfUser(
                      HttpContext, new[] { Infrastructure.Constants.ScopeMailRead });
                      return result;
            });
```

Reply:
Implemented as below...

`
        var graphClient = await GraphServiceClientFactory.GetAuthenticatedGraphClient(async () =>
            {
               string result = await tokenAcquisition.GetAccessTokenOnBehalfOfUser(
                      HttpContext, new[] { Infrastructure.Constants.ScopeMailRead });
                      return result;
            });
`